paasta_tools/cli/cmds/spark_run.py

```diff
@@ -480,10 +490,46 @@ def get_spark_conf_str(
         )
         sys.exit(1)
 
-    spark_conf = list()
-    for opt, val in dict(non_user_args, **user_args).items():
-        spark_conf.append(f'--conf {opt}={val}')
-    return ' '.join(spark_conf)
+    return dict(non_user_args, **user_args)
+
+
+def create_spark_config_str(spark_config_dict):
+    spark_config_entries = list()
+    for opt, val in spark_config_dict.items():
+        spark_config_entries.append(f'--conf {opt}={val}')
+    return ' '.join(spark_config_entries)
+
+
+def emit_resource_requirements(spark_config_dict, paasta_cluster):
+    num_executors = int(spark_config_dict['spark.cores.max']) / int(spark_config_dict['spark.executor.cores'])
+    memory_per_executor = spark_memory_to_megabytes(spark_config_dict['spark.executor.memory'])
+
+    desired_resources = {
+        'cpus': int(spark_config_dict['spark.cores.max']),
+        'mem': memory_per_executor * num_executors,
+        'disk': memory_per_executor * num_executors,  # rough guess since spark does not collect this information
+    }
+    dimensions = {'spark_framework_name': spark_config_dict['spark.app.name']}
+
+    paasta_print('Sending resource request metrics to Clusterman')
+    aws_region = get_aws_region_for_paasta_cluster(paasta_cluster)
+    metrics_client = clusterman_metrics.ClustermanMetricsBotoClient(region_name=aws_region)
+
+    with metrics_client.get_writer(clusterman_metrics.APP_METRICS) as writer:
```

Comment:
There's a possibility this call might change a little bit; since metrics get duplicated to SFX, and SFX doesn't like timeseries with unbounded dimension, we either need to filter metrics emissions that have unbounded dimension or reduce the length of time we keep this data in SFX.  I want to chat with the metrics team before making a decision here one way or the other.

Reply:
ok, I think this and the fact that I'll need to bump clusterman_metrics means I should wait to merge this